substrate/frame/scheduler/src/lib.rs

```diff
@@ -1157,24 +1190,51 @@ impl<T: Config> Pallet<T> {
 			return
 		}
 
-		let mut incomplete_since = now + One::one();
-		let mut when = IncompleteSince::<T>::take().unwrap_or(now);
-		let mut executed = 0;
+		let queue = Queue::<T>::get();
+		let end_index = match queue.iter().position(|&x| x >= now) {
```

Comment:
Why is this needed? There are only two cases: Either the first element is >= now or it is not.  
We can just check the first element in the loop below and then take it from there.